compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTupleType.java

```diff
@@ -90,7 +92,13 @@ private BTupleType(BTypeSymbol tsymbol, boolean readonly) {
 
     @Override
     public List<BType> getTupleTypes() {
-        return tupleTypes;
+        List<BType> types = new ArrayList<>();
+        memberTypes.forEach(member -> types.add(member.type));
+        return types;
+    }
```

Comment:
We should be able to cache this without creating a new list every time. The members field will have to be private (along with a new list we introduce for types) and when adding members to it/updating it we need to make sure the types are also updated.

We can do this in a separate PR once this is merged.